zeno_build/models/code_generate.py

```diff
@@ -0,0 +1,94 @@
+"""Tools to generate from prompts."""
+import asyncio
+
+import re
+import tqdm
+import torch
+import openai
+import asyncio
+import transformers
+from zeno_build.models import lm_config
+from zeno_build.prompts.prompt_utils import replace_variables
+from zeno_build.models.providers.openai_utils import (
+    generate_code_from_openai_chat_completion,
+    generate_code_from_openai_completion,
+)
+from zeno_build.models.providers.huggingface_utils import generate_code_from_huggingface
+from zeno_build.models.providers.cohere_utils import generate_code_from_cohere
+
+
+def generate_from_code_prompt(
```

Comment:
Just to clarify, would it be possible to just use `generate_from_text_prompt` as opposed to making this new function (and all the corresponding ones)? It wasn't clear to me why we need special treatment for code.
https://github.com/zeno-ml/zeno-build/blob/c59fb35baed113de66441b9f0be476f475d92b39/zeno_build/models/text_generate.py#L12

If we need to add a command line argument or two to the `generate_from_text_prompt` function to make it flexible enough to also generate code then that is fine!